File: graphview/src/main/java/de/blox/graphview/edgerenderer/StraightEdgeRenderer.java
package de.blox.graphview.edgerenderer;

import android.graphics.Canvas;
import android.graphics.Paint;

import de.blox.graphview.Edge;
import de.blox.graphview.Graph;
import de.blox.graphview.Node;
import de.blox.graphview.Vector;

/**
 *
 */
public class StraightEdgeRenderer implements EdgeRenderer {
    @Override
    public void render(Canvas canvas, Graph graph, Paint paint) {
        for (Edge edge : graph.getEdges()) {
            final Node source = edge.getSource();
            final Vector sourcePosition = source.getPosition();
            final Node destination = edge.getDestination();
            final Vector destinationPosition = destination.getPosition();

            canvas.drawLine((float) sourcePosition.getX() + source.getWidth() / 2f,
                    (float) sourcePosition.getY() + source.getHeight() / 2f,
                    (float) destinationPosition.getX() + destination.getWidth() / 2f,
                    (float) destinationPosition.getY() + destination.getHeight() / 2f, paint);
        }
    }
}

File: graphview/src/main/java/de/blox/graphview/edgerenderer/StraightEdgeRenderer.kt
package de.blox.graphview.edgerenderer

import android.graphics.Canvas
import android.graphics.Paint
import de.blox.graphview.Graph

class StraightEdgeRenderer : EdgeRenderer {
    override fun render(canvas: Canvas, graph: Graph, paint: Paint) {
        graph.edges.forEach { (source, destination) ->
            val (x1, y1) = source.position
            val (x2, y2) = destination.position

            canvas.drawLine(
                x1 + source.width / 2f,
                y1 + source.height / 2f,
                x2 + destination.width / 2f,
                y2 + destination.height / 2f, paint
            )
        }
    }
}
